clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -4304,6 +4304,37 @@ class CIRAbsOpLowering : public mlir::OpConversionPattern<cir::AbsOp> {
     return mlir::success();
   }
 };
+class CIRSignBitOpLowering : public mlir::OpConversionPattern<cir::SignBitOp> {
+public:
+  using OpConversionPattern<cir::SignBitOp>::OpConversionPattern;
+
+  mlir::LogicalResult
+  matchAndRewrite(cir::SignBitOp op, OpAdaptor adaptor,
+                  mlir::ConversionPatternRewriter &rewriter) const override {
+    assert(!::cir::MissingFeatures::isPPC_FP128Ty());
+
+    mlir::DataLayout layout(op->getParentOfType<mlir::ModuleOp>());
+    int width = layout.getTypeSizeInBits(op.getInput().getType());
+    if (auto longDoubleType =
+            mlir::dyn_cast<cir::LongDoubleType>(op.getInput().getType())) {
+      if (mlir::isa<cir::FP80Type>(longDoubleType.getUnderlying())) {
+        // see https://github.com/llvm/clangir/issues/1057
+        llvm_unreachable("NYI");
```

Comment:
For next rounds: during LLVM lowering we're better off erroring out `return op.emitError(...)` - I'm gonna merge this anyway now cause I know you're gonna be working on this soon anyways! Thanks for the fixes.